internal/pipeline/io.go

```diff
@@ -0,0 +1,92 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// Package pipeline provides struct definitions for Pipeline and utility
+// methods to read and write a pipeline resource.
+package pipeline
+
+import (
+	"io"
+	"io/ioutil"
+	"os"
+
+	"sigs.k8s.io/kustomize/kyaml/yaml"
+)
+
+const (
+	kptAPIVersion string = "kpt.dev/v1alpha1"
+	pipelineKind  string = "Pipeline"
+	defaultName   string = "pipeline"
+)
+
+var defaultSources []string = []string{"./*"}
+
+// newEmptyPipeline returns a new empty Pipeline
+func newEmptyPipeline() *Pipeline {
+	return &Pipeline{}
+}
+
+// NewPipeline returns a pointer to a new default Pipeline.
+// The default Pipeline should be:
+// apiVersion: kpt.dev/v1alpha1
+// kind: Pipeline
+// metadata:
+//   name: pipeline
+// sources:
+//   - '.*'
+func NewPipeline() *Pipeline {
+	p := newEmptyPipeline()
+	p.APIVersion = kptAPIVersion
+	p.Kind = pipelineKind
+	p.Name = defaultName
+	p.Sources = defaultSources
+	return p
+}
+
+// FromBytes returns a Pipeline parsed from bytes
+func FromBytes(b []byte) (*Pipeline, error) {
+	p := newEmptyPipeline()
+	err := yaml.Unmarshal(b, p)
+	if err != nil {
```

Comment:
General note: I see a lot of:

	if err != nil {
		return nil, err
	}

This may be valid in many places, but for each instance consider whether you can add more context on the failure instead of passing it through.